testcases/OpTestGSBStaticKey.py

```diff
@@ -128,86 +120,59 @@ def getPRePDisk(self):
-        self.assertIn(self.signature, out, fail_msg)
-
-    def checkGrub(self):
-        # Checking whether the grub is signed or not
-        cmd = "strings %s | tail -1" % self.grubFilename
-        out = self.c.run_command(cmd)
-        fail_msg = "Grub is not signed"
-        self.assertIn(self.signature, out, fail_msg)
-
-    def getRHELFiles(self):
-        '''
-        Get the signed grub file details, this is the manual step required for
-        RHEL OS
-        '''
-        cmd = "rpm -ql grub2-ppc64le"
-        out = self.c.run_command(cmd)
-        for line in out:
-            if 'core.elf' in line:
-                self.grubFilename = line
-        if not self.grubFilename:
-            self.fail("%s: Failed to get grub file" % self.distro)
-
     def os_secureboot_enable(self, enable=True):
         '''
         To enable/disable the Secure Boot at Operating System level.
         Parameter enable=True for enabling Secure Boot
         Parameter enable=False for disabling Secure Boot
         '''        
-        self.checkKernel()
-        if self.distro == 'rhel' and enable:
-            # Get the PReP disk file
-            self.getPRePDisk()
-            # Get the all the required files
-            self.getRHELFiles()
-            self.checkGrub()
-            # Running grub2-install on PReP disk
-            out = self.c.run_command("grub2-install %s" % self.prepDisk)
-            if "Installation finished. No error reported." not in out:
-                self.fail("RHEL: Failed to install on PReP partition")
-            # Copy the signed grub in to the PReP disk using 'dd' command
-            out = self.c.run_command("dd if=%s of=%s || echo 'no'" %
-                                     (self.grubFilename, self.prepDisk))
-            if "no" in "".join(out):
-                self.fail("RHEL: Failed to copy to PReP partition")
-        elif self.distro == 'rhel' and not enable:
-            # Nothing to do for Secure Boot disable for RHEL at OS level
-            pass
-        bfile = "/etc/sysconfig/bootloader"
-        if self.distro == 'sles' and enable:
-            # Add SECURE_BOOT="yes" at file  /etc/sysconfig/bootloader
-            cmd = "sed -i '/SECURE_BOOT=\"no\"/c\SECURE_BOOT=\"yes\"' %s" % bfile
-            self.c.run_command(cmd)
-            out = self.c.run_command("pbl --install || echo 'no' ")
-            if "no" in "".join(out):
-                self.fail("SLES: Failed to enable Secure Boot")
-        elif self.distro == 'sles' and not enable:
-            # Add SECURE_BOOT="no" at file  /etc/sysconfig/bootloader
-            cmd = "sed -i '/SECURE_BOOT=\"yes\"/c\SECURE_BOOT=\"no\"' %s" % bfile
-            self.c.run_command(cmd)
-            out = self.c.run_command("pbl --install || echo 'no'")
-            if "no" in "".join(out):
-                self.fail("SLES: Failed to disable Secure Boot")
+        if self.kernel_signature == "signed_kernel":
```

Comment:
What is the difference between old and new implementation. Does the new implemention work for NV environment too.

Reply:
> _What is the difference between old and new implementation._ 
The flow remains the same, but the API for checking the kernel-signature has been moved to a different location. Hence, instead of failing the test from the API, now the flow is handled in a way which would return the signature and based on that we would FAIL the test here in this file.

> _Does the new implemention work for NV environment too._
Not sure what is meant by NV environment?